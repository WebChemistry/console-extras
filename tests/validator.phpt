<?php declare(strict_types = 1);

use Symfony\Component\Console\Input\InputInterface;
use Symfony\Component\Console\Output\OutputInterface;
use Symfony\Component\Validator\Constraints\LessThan;
use Symfony\Component\Validator\ValidatorBuilder;
use Tester\Assert;
use WebChemistry\ConsoleExtras\Attribute\Argument;
use WebChemistry\ConsoleExtras\ExtraCommand;

require __DIR__ . '/bootstrap.phpt';

$command = new class extends ExtraCommand {

	#[Argument]
	#[LessThan(10)]
	protected int $num;

	public function __construct(?string $name = null)
	{
		parent::__construct($name);


		$this->setValidator(
			(new ValidatorBuilder())
				->enableAnnotationMapping()
				->getValidator()
		);
	}

	protected function exec(InputInterface $input, OutputInterface $output): bool
	{
		$output->write(implode(',', array_map(fn ($value) => var_export($value, true), [
			$this->num,
		])));

		return true;
	}

};

Assert::same("num: This value should be less than 10.\n", runConsoleApp($command, [
	'num' => '12',
]));
Assert::same('8', runConsoleApp($command, [
	'num' => '8',
]));
