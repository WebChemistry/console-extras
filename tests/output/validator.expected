num: This value should be less than 10.